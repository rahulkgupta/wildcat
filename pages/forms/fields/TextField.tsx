import { TextField } from '../../util/form';

interface Props {
    field: TextField
}


const TextFieldView = (props: Props) => {

    let field = props.field
    return (
        <div>
            <label>{field.getLabel()}</label>
            <input type="text" value={field.getValue()} onChange={(event) =>{field.setValue(event.target.value);}}></input>
        </div>
    )
}

export default TextFieldView;