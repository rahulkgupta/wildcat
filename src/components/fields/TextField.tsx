import TextField from '@src/util/form/fields/text';

interface Props {
  field: TextField;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const TextFieldView = (props: Props) => {
  const { field } = props;
  return (
    <div>
      <label>{field.getLabel()}</label>
      <input
        type="text"
        value={field.getValue()}
        onChange={(event) => {
          field.setValue(event.target.value);
        }}
        data-testid={field.getId()}
      />
    </div>
  );
};

export default TextFieldView;
