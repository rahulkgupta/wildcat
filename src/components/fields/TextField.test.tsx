import TextFieldView from './TextField';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import TextField from '@src/util/form/fields/text';

describe('TextFieldView', () => {
  test('render', () => {
    const textFieldView: ReactTestRenderer = renderer.create(
      <TextFieldView
        {...{
          field: new TextField(
            '1234',
            {
              value: 'value',
              label: 'label',
            },
            jest.fn(),
          ),
          submit: jest.fn(),
        }}
      />,
    );
    expect(textFieldView.toJSON()).toMatchSnapshot();
  });

  test('onClick', () => {
    const submit: Function = jest.fn();
    const textField = new TextField(
      '1234',
      {
        value: 'value',
      },
      jest.fn(),
    );
    const { getByTestId } = render(
      <TextFieldView
        {...{
          field: textField,
          submit,
        }}
      />,
    );
    fireEvent.change(getByTestId('1234'), { target: { value: 'new value' } });
    expect(textField.getValue()).toMatchSnapshot();
  });
});
