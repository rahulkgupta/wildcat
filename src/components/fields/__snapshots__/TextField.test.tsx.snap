// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldView onClick 1`] = `"new value"`;

exports[`TextFieldView render 1`] = `
<div>
  <label>
    label
  </label>
  <input
    data-testid="1234"
    onChange={[Function]}
    type="text"
    value="value"
  />
</div>
`;
